src/main/kotlin/nl/connectplay/scoreplay/data/DatabaseNotificationRepository.kt

```diff
@@ -1,11 +1,122 @@
 package nl.connectplay.scoreplay.data
 
+import kotlinx.coroutines.coroutineScope
 import nl.connectplay.scoreplay.abstraction.data.NotificationRepository
 import nl.connectplay.scoreplay.models.dto.notifications.NewNotificationDto
+import nl.connectplay.scoreplay.models.dto.notifications.NotificationDto
+import kotlinx.coroutines.async
+import java.util.UUID
 
 class DatabaseNotificationRepository(private val database: Database) : NotificationRepository {
     override suspend fun saveNotificationAsync(notification: NewNotificationDto) {
         TODO("Not yet implemented")
     }
 
-}
\ No newline at end of file
+    override suspend fun getNotificationByIdAsync(notificationId: UUID, userId: Int): NotificationDto? = coroutineScope {
+        async {
+            database.connection?.use { connection ->
+                val sql = """SELECT
+                                notification_id,
+                                content,
+                                read
+                             FROM notifications
+                                score_id = ?
+                          """.trimIndent()
+
+                val stmt = connection.prepareStatement(sql)
+                stmt.setString(1, notificationId.toString())
+
+                val resultSet = stmt?.executeQuery()
+                var notification: NotificationDto? = null;
+                if (resultSet?.next() == true) {
+                    notification = NotificationDto(
+                        notificationId = UUID.fromString(resultSet.getString("notification_id")),
+                        content = resultSet.getString("content"),
+                        read = resultSet.getBoolean("read")
+                    )
+                }
+
+                stmt?.close()
+                resultSet?.close()
+
+                notification
+            }
+        }.await()
+    }
+
+    override suspend fun getAllNotificationsAsync(limit: Int?, offset: Int? ,userId: Int): List<NotificationDto>? {
```

Comment:
limit en offset hoeven niet nullable te zijn. pas dat dan ook aan in je interface